Dockerfile

```diff
@@ -5,13 +5,32 @@ RUN sed -i 's/ universe/ universe multiverse/' /etc/apt/sources.list
 
 RUN apt-get update && \
   apt-get dist-upgrade --yes && \
-  apt-get install --yes \
-  git \
-  snapcraft \
-  && \
   apt-get autoclean --yes && \
   apt-get clean --yes
 
+# Install snapcraft dependencies
+RUN apt-get install git sudo libslang2-dev python-pip python3-pip gcc g++ make python3-dev python3-venv libffi-dev \
+  libsodium-dev libapt-pkg-dev libarchive13 squashfs-tools patchelf --yes && \
+  apt-get autoclean --yes && \
+  apt-get clean --yes
+
+ADD ./ /snapcraft
+
+RUN mkdir -p /venv/snapcraft && \
+python3 -m venv /venv/snapcraft && \
+. /venv/snapcraft/bin/activate && \
+pip --no-cache-dir install wheel && \
+pip --no-cache-dir install -r /snapcraft/requirements.txt
+
+RUN echo """#!/bin/bash\n\
+    \n\
+    . /venv/snapcraft/bin/activate\n\
+    python3 /snapcraft/bin/snapcraft \$@\n\
+    deactivate\
+    """ > /bin/snapcraft && chmod +x /bin/snapcraft
```

Comment:
This could just be a file in the repo that you copy into place

Reply:
:+1: